### ISSUE TITLE
Generic instance

### ISSUE THREAD

**Issue by akegalj** [rgrempel/purescript-int-53#5]
Is there a reason of not heaving default derived `Generic` instance ?

**Comment by rgrempel** [comment]
Generally speaking, having a default `Generic` instance is equivalent to exporting your constructors, so you might not want one for the same reasons you might not want to export your constructors.

For instance, in this case, the `Int53` constructor takes a `Number` internally. However, we don't export it, because not all numbers are integers (we just use `Number` as a convenience to get all 53 bits). Since we want to limit `Int53` values to actual integers, we can't export the constructor.

If we had a default `Generic` instance, you could do something like this:

``` purescript
badInt53 = fromSpine (SProd "Data.Int53.Int53" [\_ -> SNumber 27.3]) :: Maybe Int53
-- Just (Int53 27.3)
```

... which we wouldn't want to be possible.

Now, it's possible that we could write a `Generic` instance manually that would be fine ... it would probably use `fromNumber` at some point to take an `SNumber` and turn it into a `Maybe Int53`. But I haven't entirely thought that through.

**Comment by akegalj** [comment]
> However, we don't export it, because not all numbers are integers
> I agree, didn't think about it enough.

Yes, we are using custom `Generic` instance, but I guess I was slightly annoyed with writing additional newtype wrapper so that I can define `Generic`. But you are right, adding generic would be wrong in this case. Thank you for your explanation.

Note that you can derive `Eq` and `Ord` . I think their derived instance is the same as your manual instances.

**Comment by rgrempel** [comment]
Actually, I was thinking of adding a manual `Generic` instance to `Int53` itself, so you wouldn't need the `newtype`. Something like this:

https://github.com/rgrempel/purescript-int-53/commit/ea725503908afabf5a417fe2e3770ae3bbe31740

Does that look like it might be helpful? I can merge it if so.

**Comment by akegalj** [comment]
We are using more simple instance of `Generic` (it might be wrong general case but it suits our needs). I don't have time now so will review it tomorrow (and read more about `Generic`, don't have much experience with it).

**Comment by rgrempel** [comment]
I've published version 2.1.0, with the manually-written `Generic` instance.

**Comment by akegalj** [comment]
Thank you. Sorry for not being able to review your code (I had it on my todo but can't allocate free time currently). Many thanks <3
